core/primitives/src/bandwidth_scheduler.rs

```diff
@@ -129,6 +129,39 @@ impl BandwidthRequest {
 
         Some(BandwidthRequest { to_shard, requested_values_bitmap: bitmap })
     }
+
+    /// Create a basic bandwidth request when receipt sizes are not available.
+    /// It'll request two values - one corresponding to the first receipt in
+    /// the outgoing buffer and one corresponding to max_receipt_size.
+    /// This ensures that the blockchain will be able to make progress while
+    /// waiting for the receipt sizes to become available.
+    /// The resulting behaviour is similar to the previous approach with allowed shard.
+    pub fn make_basic(
+        to_shard: u8,
```

Comment:
I was trying to save space, since the requests are in chunk header and we could have num_shards**2 of them in the worst case. Switching to u16 when the time comes shouldn't be too hard, `BandwidthRequests` is a versioned struct.
It would be nice to have something like "compact int representation" in borsh, e.g https://reference.cash/protocol/formats/variable-length-integer

Reply:
Alright but I'm still not a fan, especially in light of dynamic resharding in the future where the shard ids will be out of our control. Can you add a test like this one - `test_shard_id_u16_optimization` - but for u8?